flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/TaskManagerRegistration.java

```diff
@@ -68,7 +68,9 @@ public boolean isIdle() {
 	}
 
 	public void markIdle() {
-		idleSince = System.currentTimeMillis();
+		if (!isIdle()) {
+			idleSince = System.currentTimeMillis();
+		}
```

Comment:
The reason that the SM idleness unit tests didn't catch this is, `markIdle` is called when a slot report is received, which in practice occurs continuously but is tough to simulate.  Just making clear that there are idle tests but I didn't see an obvious way to improve them.